src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -396,29 +273,41 @@ public Completable start() {
                     }
                 }
 
+                connectionState.transport = transport;
+
                 transport.setOnReceive(this.callback);
                 transport.setOnClose((message) -> stopConnection(message));
 
                 return transport.start(negotiateResponse.getFinalUrl()).andThen(Completable.defer(() -> {
                     ByteBuffer handshake = HandshakeProtocol.createHandshakeRequestMessage(
-                            new HandshakeRequestMessage(protocol.getName(), protocol.getVersion()));
-
-                    connectionState = new ConnectionState(this);
-
-                    return transport.send(handshake).andThen(Completable.defer(() -> {
-                        timeoutHandshakeResponse(handshakeResponseTimeout, TimeUnit.MILLISECONDS);
-                        return handshakeResponseSubject.andThen(Completable.defer(() -> {
-                            hubConnectionStateLock.lock();
+                                new HandshakeRequestMessage(protocol.getName(), protocol.getVersion()));
+
+                    return connectionState.transport.send(handshake).andThen(Completable.defer(() -> {
```

Comment:
Do we need the lock to call connectionState.transport.send()? I don't think we have the lock here.

Reply:
I don't think so, we have the `connectionState` so there is no risk of it being null.